Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -101,6 +105,53 @@ public class RestRequest {
         try? session.syncShutdown()
     }
 
+    fileprivate static var _globalELG: EventLoopGroup?
+
+    /// Provides access to the global EventLoopGroup used by all instances of RestRequest.
+    /// This can be set once (and once only) by calling `RestRequest.setGlobalELG(elg:)`
+    /// before creating a request.
+    ///
+    /// Defaults to a `MultiThreadedEventLoopGroup` containing 1 thread per available
+    /// processor.
+    ///
+    /// Note that the default value is assigned lazily, and you cannot call `setGlobalELG` after
+    /// accessing this property.
+    public private(set) static var globalELG: EventLoopGroup {
+        get {
+            if let result = _globalELG { return result }
+            #if os(Linux)
+            let numberOfCores = Int(linux_sched_getaffinity())
+            let result = MultiThreadedEventLoopGroup(numberOfThreads: numberOfCores > 0 ? numberOfCores : System.coreCount)
+            #else
+            let result = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
+            #endif
+            _globalELG = result
+            return result
+        }
+        set {
+            _globalELG = newValue
```

Comment:
This looks racey....

Reply:
How about https://github.com/IBM-Swift/SwiftyRequest/pull/78/commits/b96f1869acc602ee9f54528ca52da3afbecfc771